src/utils_db_query.c

```diff
@@ -1011,7 +1012,13 @@ int udb_query_prepare_result (udb_query_t const *q, /* {{{ */
   if ((q == NULL) || (prep_area == NULL))
     return (-EINVAL);
 
-  udb_query_finish_result (q, prep_area);
```

Comment:
>  It's generally a good idea to make sure you clean up after yourself

That is right, but in this case (that is udb_query_prepare_result() function code) udb_query_finish_result() is called _before_, not _after_.
(